File: PROGRAMMERS/LEVEL1/실패율.cpp
#include <string>
#include <algorithm>
#include <vector>

using namespace std;
int arr[501] = { 0, };

bool cmp(pair<double, int>& a, pair<double, int>& b) {
    if (a.first == b.first) return a.second < b.second;
    else return a.first > b.first;
}

vector<int> solution(int N, vector<int> stages) {
    vector<int> answer;
    int total = stages.size();
    vector<pair<double, int>> v;

    vector<int> result;

    for (int i = 0; i < stages.size(); i++) {
        arr[stages[i]]++;
    }

    for (int i = 1; i <= N; i++) {
        if (total == 0) v.push_back({ 0,i });
        else {
            v.push_back({ (double)arr[i] / total, i });
            total -= arr[i];
        }
    }

    sort(v.begin(), v.end(), cmp);

    for (int i = 0; i < N; i++) {
        result.push_back(v[i].second);
    }

    return result;
}

File: PROGRAMMERS/LEVEL1/실패율.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

class Solution {
    public int[] solution(int N, int[] stages) {
        int[] answer = new int[N];
        double total = stages.length;
        List<double[]> fail = new ArrayList<>();
        
        int cur=0;
        
        for(int i=1;i<=N;i++) {
        	for(int j=0;j<stages.length;j++) {
        		if(i==stages[j]) {
        			cur++;
        		}
        	}
        	if(cur==0) {
        		fail.add(new double[]{i,0});
        	}
        	else {
        		fail.add(new double[] {i,cur/total});
        		total-=cur;
        		cur=0;
        	}
        }
        
        fail.sort((o1,o2)->Double.compare(o2[1], o1[1]));
        
        for(int i=0;i<fail.size();i++) {
        	answer[i]=(int)fail.get(i)[0];
        }
        
        return answer;
    }
}